functions/callback-functions.md

```diff
@@ -0,0 +1,214 @@
+We could approach the third argument by expecting a string, such as `'add'`, `'+'`, `'multiply'`, or `'*'`.
+
+<!-- prettier-ignore-start -->
+```js
+const calculator = function(a, b, operation){
+  // if (operation === 'add') {
+  //  return a + b;
+  // }
+}
+```
+<!-- prettier-ignore-end -->
+
+However, wouldn't it be grand if we could _pass in_ the actual operation, whether it's addition or subtraction, and execute it?
+
+## Callback Functions
+
+In order to pass around behavior, we pass around functions.
+
+More specifically, a function can _take in a function_ as an _argument_, and then call that function.
+
+To help us talk about functions, we can use specific vocabulary around these functions. A **callback function** is a nickname for a function that is _passed into_ another function.
+
+We can reverse that definition and imagine a function that _takes in another function as an argument_. The function passed in as an argument can be labeled as a _callback function_.
+
+### Passing Around Cleaning Up Behavior
+
+Let's dive into an example! Let's define these two functions, `cleanKitchen` and `bakeCake`.
+
+<!-- prettier-ignore-start -->
+```js
+const cleanCountertop = function () {
+  console.log('I\'m cleanin\' my counters now! Scrubba scrub scrub.');
+}
+
+const bakeCake = function () {
+  console.log('Make the batter...');
+  console.log('Put it in the oven...');
+}
+
+bakeCake();
+```
+<!-- prettier-ignore-end -->
+
+The variable `cleanCountertop` references a function that prints details about cleaning a kitchen.
+
+The variable `bakeCake` references a different function that prints details about baking a cake.
+
+When we bake a cake, maybe there's always cleanup behavior that we want to call, whether it's cleaning our counters, sink, or stove-top.
+
+We can _change the function signature_ of `bakeCake` to accept an argument, `cleanupBehavior`. Inside our `bakeCake` function, we can take that argument and _invoke_ whatever function it references, by attaching `()` to it.
+
+<!-- prettier-ignore-start -->
+```js
+const bakeCake = function (cleanupBehavior) {
+  console.log('Make the batter...');
+  console.log('Put it in the oven...');
+  cleanupBehavior();
+}
+```
+<!-- prettier-ignore-end -->
+
+We can refer to `cleanupBehavior` as a _callback function_ inside our `bakeCake` function. After we look at the big picture, we can think about the phrase "callback."
+
+<!-- prettier-ignore-start -->
+```js
+const cleanCountertop = function () {
+  console.log('I\'m cleanin\' my counters now! Scrubba scrub scrub.');
+}
+
+const bakeCake = function (cleanupBehavior) {
+  console.log('Make the batter...');
+  console.log('Put it in the oven...');
+  cleanupBehavior();
+}
+
+bakeCake(cleanCountertop);
+```
+<!-- prettier-ignore-end -->
+
+When we ultimately _invoke_ the `bakeCake` function with `bakeCake()`, we need to pass in an argument.
+
+<!-- prettier-ignore-start -->
+```js
+bakeCake(cleanCountertop);
+```
+<!-- prettier-ignore-end -->
+
```

Comment:
Include a log of the output.
Actual output markdown in following comment.
Be careful to follow the log with a blank line (can't add it in the comment)
Not part of this suggestion because code.
```suggestion

This would produce the following output:

```

Reply:
```
Make the batter...
Put it in the oven...
I'm cleanin' my counters now! Scrubba scrub scrub.
Take it out...
Om nom nom!!!
```